assembly/pom.xml

```diff
@@ -75,11 +75,7 @@
       <version>${project.version}</version>
     </dependency>
 
-    <!--
-      Because we don't shade dependencies anymore, we need to restore Guava to compile scope so
-      that the libraries Spark depend on have it available. We'll package the version that Spark
-      uses (14.0.1) which is not the same as Hadoop dependencies, but works.
-    -->
+    <!-- Restore Guava to compile scope so that the libraries Spark depend on have it available -->
```

Comment:
the existing comments are outdated, Hadoop already switched to `hadoop-shaded-guava-1.2.0.jar`

Reply:
One additional question is should we remove `<scope>${hadoop.deps.scope}</scope>`?

Reply:
Please open a new JIRA and PR to discuss this.

Reply:
opened SPARK-48255